src/seedu/addressbook/data/person/PrivatableContactDetail.java

```diff
@@ -0,0 +1,21 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * 
+ * Represents a contact detail that can be private
+ *
+ */
+public abstract class PrivatableContactDetail extends ContactDetail {
```

Comment:
this is good and thought through; however, i personally feel this is an overkill for this exercise.
good to see that you are making full use of the OO power :-)